doc/introduction.md

```diff
@@ -14,3 +14,97 @@ The essential concepts in RxJS which solve async event management are:
 - **Operators:** are pure functions that enable a functional programming style of dealing with collections with operations like `map`, `filter`, `concat`, `flatMap`, etc.
 - **Subject:** is the equivalent to an EventEmitter, and the only way of multicasting a value or event to multiple Observers.
 - **Schedulers:** are centralized dispatchers to control concurrency, allowing us to coordinate when computation happens on e.g. `setTimeout` or `requestAnimationFrame` or others.
+
+## First examples
+
+Normally you register event listeners.
+```js
+const button = document.querySelector('button');
+button.addEventListener('click', () => console.log('Clicked!'));
+```
+
+Using RxJS you create an observable instead.
+```js
+const button = document.querySelector('button');
+Rx.Observable.fromEvent(button, 'click')
+  .subscribe(() => console.log('Clicked!'));
+```
+
+
+## Purity
+What makes RxJS powerful is its ability to produce values using pure functions. That means your code is less prone for errors.
+
+Normally you would create an impure function, where other
+pieces of your code can mess up your state.
+```js
+let count = 0;
+const button = document.querySelector('button');
+button.addEventListener('click', () => console.log(`Clicked ${++count} times`));
+```
+
+Using RxJS you isolate the state.
+```Js
+const button = document.querySelector('button');
+Rx.Observable.fromEvent(button, 'click')
+  .scan(count => count + 1, 0)
+  .subscribe(count => console.log('Clicked ${count} times'));
+```
+
+The **scan** operator works just like **reduce**. It takes a value which is exposed to a callback. The returned value of the callback will then become the next value exposed the next time the callback runs.
+
+## Flow
+RxJS has a whole range of operators that helps you control how the events flow through your observables.
+
+Only allow clicks every second with plain JS.
+```js
+let count = 0;
+let lastClick = null;
+const button = document.querySelector('button');
+button.addEventListener('click', () => {
+  if (lastClick && Date.now() - lastClick >= 1000) {
+    console.log(`Clicked ${++count} times`);  
+  } else {
+    lastClick = Date.now();
+  }
+});
+```
+
+With RxJS.
+```js
+const button = document.querySelector('button');
+Rx.Observable.fromEvent(button, 'click')
+  .throttle(1000)
+  .scan(count => count + 1, 0)
+  .subscribe(count => console.log('Clicked ${count} times'));
+```
+
+Other flow control operators are **filter**, **delay**, **debounce**, **take**, **takeUntil**, **distinct**, **distinctUntilChanged** etc.
+
+## Values
+You can change the values passed through your observables.
+
+Add count with x position of mouse with plain JS
```

Comment:
"`Here's how you can add the current mouse x position for every click, in plain JavaScript:`"